doozer/doozerlib/cli/scan_osh.py

```diff
@@ -1,29 +1,289 @@
+                 skip_diff_check: Optional[bool]):
         self.runtime = runtime
         self.last_brew_event = last_brew_event
         self.dry_run = dry_run
         self.brew_tags = []
         self.specific_nvrs = nvrs
         self.check_triggered = check_triggered
         self.all_builds = all_builds
+        self.create_jira_tickets = create_jira_tickets
         self.error_nvrs = []
+        self.version = self.runtime.group.split("-")[-1]
+        self.jira_project = "OCPBUGS"
+        self.skip_diff_check = skip_diff_check
 
         # Initialize runtime and brewhub session
         self.runtime.initialize(clone_distgits=False)
         self.koji_session = koji.ClientSession(self.runtime.group_config.urls.brewhub)
 
+        # Initialize JIRA client
+        self.jira_client: JIRA = self.runtime.build_jira_client()
+
+    def get_scan_info(self, brew_package_name: str):
+        """
+        Find the last successful scan for a particular package and retrieve its ID
+        """
+        # Eg: osh-cli find-tasks --regex ose-network-tools-container-v4.15
+        cmd = f"osh-cli find-tasks --regex {brew_package_name}"
+
+        _, result, _ = cmd_gather(cmd)
+
+        finished_tasks = result.strip().split("\n")
+
+        for index, task_id in enumerate(finished_tasks):
+            cmd = f"osh-cli task-info {task_id}"
+            self.runtime.logger.info(f"Running: {cmd}")
+
+            _, result, _ = cmd_gather(cmd)
+
+            if "state_label = CLOSED" in result:
+                try:
+                    previous_scan_id = finished_tasks[index + 1]
+
+                    cmd = f"osh-cli task-info {previous_scan_id}"
+                    self.runtime.logger.info(f"Running: {cmd}")
+                    _, result, _ = cmd_gather(cmd)
+
+                    previous_scan_nvr = yaml.safe_load(result.replace(" =", ":"))["label"]  # returns NVR
+
+                    return task_id, previous_scan_id, previous_scan_nvr
+                except IndexError:
+                    return task_id, None, None
+
+    @staticmethod
+    def check_if_vulnerabilities_exist(task_id: str, nvr: str) -> bool:
+        """
+        ProdSec only wants us to check the scan-results-imp.js file of a scan and see if vulnerabilities are reported.
+        """
+
+        response = requests.get(SCAN_RESULTS_URL_TEMPLATE.format(task_id=task_id, nvr=nvr))
+
+        if len(response.json()["defects"]) > 0:
+            return True
+
+        return False
+
+    def get_distgit_name_from_brew_nvr(self, nvr: str) -> str:
+        """
+            Returns the distgit name from the brew nvr
+
+            :param nvr: Full NVR eg: openshift-enterprise-pod-container-v4.14.0-202310031045.p0.g9ef6de6.assembly.stream
+            """
+        brew_package_name = self.koji_session.getBuild(buildInfo=nvr, strict=True)["package_name"]
+
+        _, output, _ = cmd_gather(f"doozer --disable-gssapi -g {self.runtime.group} "
```

Comment:
if this function gets called for each nvr, it's better if we cache this command output

Reply:
Do we already have a cache header anywhere or shall I create one?

Reply:
I'm not sure what you mean by cache header .. I was thinking of a var that stores this mapping the first time this method is called and then on subsequent calls we return from it

Reply:
I meant something like [this](https://github.com/openshift-eng/art-bot/blob/9dd835fa6b1c76a1f9f2f7d4d0bfdccdefd0b3f4/artbotlib/util.py#L77)

Reply:
But what you mentioned is the best way I feel, implemented in https://github.com/openshift-eng/art-tools/pull/141/commits/ce0bb7167b9640df3b86cbe1e497a419585d656a

Reply:
Instead of calling it in the constructor always, I would suggest we move it to the func itself, so
```python
# in constructor
self.brew_distgit_mapping = None

def get_brew_distgit_mapping(self):
    if self.brew_distgit_mapping:
        return self.brew_distgit_mapping[brew_package_name]
    # else call doozer and store
```